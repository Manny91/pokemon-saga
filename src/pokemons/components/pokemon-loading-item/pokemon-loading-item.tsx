import React from "react";
import { Loading, LoadingDot } from "../loading/loading";
import styled from "../../../styled-components";

export const PokemonLoadingItem = () => {
  return (
    <LoadingWrapper>
      <Loading />
    </LoadingWrapper>
  );
};

const LoadingWrapper = styled.div`
  ${LoadingDot} {
    width: 15px;
    height: 15px;
  }
`;
