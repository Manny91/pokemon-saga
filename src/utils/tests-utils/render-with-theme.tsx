import React from "react";
import { ThemeProvider } from "../../styled-components";
import { theme } from "../../theme";
import { render } from "@testing-library/react";

export const renderWithTheme = (ui: JSX.Element) => {
  return render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
};
