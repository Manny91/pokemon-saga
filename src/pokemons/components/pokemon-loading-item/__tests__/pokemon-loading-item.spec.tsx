import React from "react";
import { PokemonLoadingItem } from "../pokemon-loading-item";
import "jest-styled-components";
import { renderWithTheme } from "../../../../utils/tests-utils/render-with-theme";

describe("PokemonLoadingItem Component", () => {
  it("should render properly", () => {
    const subject = renderWithTheme(<PokemonLoadingItem />);
    const tree = subject.container;
    expect(tree).toMatchSnapshot();
  });
});
