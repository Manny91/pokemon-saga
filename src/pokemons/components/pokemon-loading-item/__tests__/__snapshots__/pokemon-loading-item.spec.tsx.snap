// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonLoadingItem Component should render properly 1`] = `
.c3 {
  width: 4px;
  height: 4px;
  background-color: black;
  border-radius: 1px;
  margin: 0px 2px;
  list-style-type: none;
  -webkit-animation: fZpxSO 1s infinite ease-in-out both;
  -webkit-animation: fZpxSO 1s infinite ease-in-out both;
  animation: fZpxSO 1s infinite ease-in-out both;
}

.c1 {
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0px;
}

.c1 .c2:nth-child(1) {
  -webkit-animation-delay: -0.35s;
  animation-delay: -0.35s;
}

.c1 .c2:nth-child(2) {
  -webkit-animation-delay: -0.2s;
  animation-delay: -0.2s;
}

.c0 .c2 {
  width: 15px;
  height: 15px;
}

<div>
  <div
    class="c0"
  >
    <ul
      class="c1"
    >
      <li
        class="c2 c3"
      />
      <li
        class="c2 c3"
      />
      <li
        class="c2 c3"
      />
    </ul>
  </div>
</div>
`;
